Applications/Hermes/hermesflaskserver.py

```diff
@@ -0,0 +1,177 @@
+    """
+    # Global variables
+    global dictmsgobj
+
+    # If POST
+    if request.method == 'POST':
+        # Parse Flask arguments
+        localcallsign = request.args.get("localcallsign").upper()
+        localnodeid = request.args.get("localnodeid")
+        destinationcallsign = request.args.get("destinationcallsign").upper()
+        destinationnodeid = request.args.get("destinationnodeid")
+        data = request.args.get("data")
+
+        # Get hermes TX/RX object from global dictionary
+        unitmsgobj = dictmsgobj[localcallsign + '-' + localnodeid]
+
+        # Transmit data to remote device
+        unitmsgobj.transmit.send(destinationcallsign, int(destinationnodeid), str(data))
+
+        # Return status
+        return json.dumps(
+            {"status": "Posted Packet(s)"
+             }), 200
+
+    # If GET
+    else:
+        # Parse Flask arguments
+        localcallsign = request.args.get("localcallsign").upper()
+        localnodeid = request.args.get("localnodeid")
+
+        # Get hermes TX/RX object from global dictionary
+        unitmsgobj = dictmsgobj[localcallsign + '-' + localnodeid]
+
+        # Get next message from RX queue
+        received_item = unitmsgobj.receive.getqueueitem()
+
+        # Pickle dictionary data
+        received_data_pickle = cPickle.dumps(received_item)
+
+        # Encode to BASE64
+        received_item_64 = base64.b64encode(received_data_pickle)
+
+        return json.dumps(received_item_64, indent=1), 200, \
+            {'Content-Type': 'application/json'}
+
+
+@app.route('/queue', methods=['GET'])
+def getqueue():
+    """
+    Flask function that returns the size of the receiver queue of the specified local unit.
+
+    GET:
+
+    :param localcallsign: Flask parameter - Local unit callsign to transmit from
+    :param localnodeid: Flask parameter - Local unit node ID to transmit from
+
+    :return:
+    """
+    # Global variables
+    global dictmsgobj
+
+    # Parse Flask arguments
+    localcallsign = request.args.get("localcallsign").upper()
+    localnodeid = request.args.get("localnodeid")
+    unitmsgobj = dictmsgobj[localcallsign + '-' + localnodeid]
+
+    """This function returns the number of packets in the receiver queue."""
+    # Check Queue size of Unit #2 and receive packet (if received due to non-ARQ protocol)
+    data = {"queuesize": unitmsgobj.receive.getqueuesize()}
+
+    # Pickle dictionary data
+    data_pickle = cPickle.dumps(data)
+
+    # Encode to BASE64
+    data_64 = base64.b64encode(data_pickle)
+
+    return json.dumps(data_64, indent=1), 200, \
+           {'Content-Type': 'application/json'}
+
+
+def main():
+    """Main function which starts the Flask server."""
+    # Global variables
+    global dictmsgobj
+
+    # Get INI file configuration for Flask server
+    config_host = config.get("FLASK", "HOST")
+    config_port = config.get("FLASK", "PORT")
+
+    # Start the flask server on localhost:8001
```

Comment:
No, key is the key item as a string:

![image](https://cloud.githubusercontent.com/assets/339144/24601235/f44316c2-180c-11e7-9769-c6e0c9f4de89.png)


![image](https://cloud.githubusercontent.com/assets/339144/24601231/f01bf848-180c-11e7-97b3-1faf8ccd3f4b.png)